lib/wprof/gen_prof.rb

```diff
@@ -0,0 +1,62 @@
+class GenProf
+  def initialize(event, rec_type)
+    @event = event
+    @db_runtime = WProf::Config.get_value(:db_runtime)
+    @async_report = WProf::Config.get_value(:async)
+    generate_profiling(rec_type)
+  end
+
+  def generate_profiling(rec_type)
+    common_params
+    case rec_type
```

Comment:
We can move this `if/case` statement to it's own method so it's not all clutter, so the whole method looks something like this:
```
def generate_profiling(rec_type)
  generate_common_params
  generate_custom_params(rec_type)
  deploy_wprof_reporter
end
```

Reply:
Done.